client/web/src/site-admin/SiteAdminConfigurationPage.tsx

```diff
@@ -526,3 +543,142 @@ class SiteAdminConfigurationContent extends React.Component<Props, State> {
+            [0, 0]
+        )
+
+const SiteConfigurationHistoryItem: React.FunctionComponent<SiteConfigurationHistoryItemProps> = ({ node }) => {
+    const [open, setOpen] = useState<boolean>(false)
+
+    const icon = open ? mdiChevronUp : mdiChevronDown
+
+    if (node.reproducedDiff) {
+        const [removedLines, addedLines] = node.diff ? linesChanged(node.diff) : [0, 0]
+
+        return (
+            <Collapse key={node.id} isOpen={open} onOpenChange={setOpen}>
+                <CollapseHeader
+                    as={Button}
+                    aria-expanded={open}
+                    type="button"
+                    className="d-flex p-0 justify-content-start w-100"
+                >
+                    <Icon aria-hidden={true} svgPath={icon} />
+                    <span>
+                        <Timestamp date={node.createdAt} />
+                    </span>{' '}
+                    {node.author && (
+                        <span className="ml-1">
+                            by{' '}
+                            <Link to={`/users/${node.author.username}`} className="text-truncate">
+                                {node.author.displayName}
+                            </Link>
+                        </span>
+                    )}
+                    {node.diff && (
+                        <span className="ml-auto">
+                            <DiffStatStack className="mr-1" added={addedLines} deleted={removedLines} />
+                        </span>
+                    )}
+                </CollapseHeader>
+                <CollapsePanel>
+                    <Code className={classNames('p-2', 'mt-2', styles.diffblock)}>{node.diff}</Code>
+                </CollapsePanel>
+            </Collapse>
+        )
+    }
+    return (
+        <CollapseHeader className="d-block mb-3">
+            <>
+                {node.author} {node.createdAt}
+            </>
+        </CollapseHeader>
+    )
+}
+
+const SITE_CONFIGURATION_CHANGE_CONNECTION_QUERY = gql`
+    query SiteConfigurationHistory($first: Int, $last: Int, $after: String, $before: String) {
+        site {
+            __typename
+            configuration {
+                history(first: $first, last: $last, after: $after, before: $before) {
+                    __typename
+                    totalCount
+                    nodes {
+                        __typename
+                        ...SiteConfigurationChangeNode
+                    }
+                    pageInfo {
+                        hasNextPage
+                        hasPreviousPage
+                        endCursor
+                        startCursor
+                    }
+                }
+            }
+        }
+    }
+
+    fragment SiteConfigurationChangeNode on SiteConfigurationChange {
+        id
+        author {
+            id
+            username
+            displayName
+        }
+        reproducedDiff
+        diff
+        createdAt
+        updatedAt
```

Comment:
some fields doesn't seem to be used
```suggestion
        id
        author {
            username
            displayName
        }
        reproducedDiff
        diff
        createdAt
```